File: app/api/schemas/jms.py
from pydantic import BaseModel
from typing import Optional

from wisp.protobuf.common_pb2 import RiskLevel


class CommandRecord(BaseModel):
    input: Optional[str] = None
    output: Optional[str] = None
    risk_level: str = RiskLevel.Normal


class JMSState(BaseModel):
    id: str
    activate_review: Optional[bool] = None
    new_dialogue: Optional[bool] = None


class Conversation(BaseModel):
    id: str

File: pkg/schemas/jms.go
package schemas

import "github.com/jumpserver/wisp/protobuf-go/protobuf"

type CommandRecord struct {
	Input     string             `json:"input,omitempty"`
	Output    string             `json:"output,omitempty"`
	RiskLevel protobuf.RiskLevel `json:"risk_level"`
}
type ReviewState int

const (
	Wait ReviewState =  -1 + iota
	Rejected
	Approve
)

type JMSState struct {
	ID             string      `json:"id"`
	ActivateReview ReviewState `json:"activate_review,omitempty"`
	NewDialogue    bool        `json:"new_dialogue,omitempty"`
}

type Conversation struct {
	ID string `json:"id"`
}
